src/editor/Core/Configuration.re

```diff
@@ -4,20 +4,35 @@
  * Configuration settings for the editor
  */
 
-type tablineMode =
-  | Buffers
-  | Tabs
-  | Hybrid;
+[@deriving (show({with_path: false}), yojson)]
+type editorTablineMode =
+  | [@name "buffers"] Buffers
+  | [@name "tabs"] Tabs
+  | [@name "hybrid"] Hybrid;
 
+[@deriving (show({with_path: false}), yojson({strict: false, exn: true}))]
 type t = {
   editorLineNumbers: LineNumber.setting,
   editorMinimapEnabled: bool,
-  tablineMode,
+  editorTablineMode,
 };
 
-let create: unit => t =
-  () => {
-    editorLineNumbers: On,
-    editorMinimapEnabled: true,
-    tablineMode: Buffers,
-  };
+let ofFile = filePath => Yojson.Safe.from_file(filePath) |> of_yojson_exn;
+
+let getConfigPath = () => {
+  let sep = Filename.dir_sep;
+  Revery_Core.Environment.getWorkingDirectory()
```

Comment:
figure this will be an issue :laughing: cc @bryphe. I tried using a similar strategy to other paths like `neovimPath` but that's a tough one since those values are pulled in after the app has started whereas currently the config is expected earlier than that, could always refactor things to be an option that is populated after intialization but tbh thought this could be different (I don't recall why we needed the other approach :older_man:)

Reply:
Ignore all of that, tired brain. I now derive the path from the setup but that isn't an issue at all actually :+1: